solutions/USACO open19-milk-factory.mdx

```diff
@@ -0,0 +1,84 @@
+---
+id: usaco-940
+title: USACO 2019 December Bronze - Milk Factory
+author: Jesse Choe
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_factory_bronze_open19.html)
+
+This problem can also be solved using Depth-First Search (a Silver topic). First, I must represent the graph as a directed unweighted graph. By setting $b_i$ as the starting node and $a_i$ as the ending node, we can recursively traverse this graph across each edge list for every node from $1 ... N$. With a boolean array, we can keep track of whether each node has been visited for the currently checked node. If it has, then we can print that node, otherwise, we print -1 if the visited array is never full for each node we traverse.
```

Comment:
> we can recursively traverse this graph across each edge list for every node from $1 ... N$

Just "dfs for each node."

Reply:
It's not clear that you are running $N$ DFSs (one starting at each node) rather than one DFS that reaches every node.

Reply:
Also you didn't write $-1$

Reply:
Why did you remove the link to DFS :/